pkg/agent/client.go

```diff
@@ -439,6 +450,10 @@ func (a *Client) Serve() {
 				dialResp.GetDialResponse().ConnectID = connID
 				if err := a.Send(dialResp); err != nil {
 					klog.ErrorS(err, "could not send dialResp")
+					// clean-up is normally called from remoteToProxy which we will never invoke.
+					// So we are invoking it here to force the clean-up to occur.
+					closeOnce.Do(func() { close(dialDone) })
```

Comment:
What is the point of `dialDone`? Can we just get rid of it?

It looks like cleanup can be called in 3 places:
1. Here, in the same go routine (doesn't need the channel)
2. `remoteToProxy` - likely started after the channel is closed
3. From a CLOSE_REQ, but a close_req for the connectionID could only be received after the DIAL_RSP was already sent?

Reply:
cleanup is also called from https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/agent/client.go#L351.